src/load_dotenv.rs

```diff
@@ -1,45 +1,66 @@
 use crate::common::*;
 
+const DEFAULT_DOTENV_FILENAME: &str = ".env";
+
 pub(crate) fn load_dotenv(
   config: &Config,
   settings: &Settings,
   working_directory: &Path,
 ) -> RunResult<'static, BTreeMap<String, String>> {
-  // `dotenv::from_path_iter` should eventually be un-deprecated, see:
-  // https://github.com/dotenv-rs/dotenv/issues/13
-  #![allow(deprecated)]
-
   if !settings.dotenv_load.unwrap_or(true) {
```

Comment:
Okay, last thing I swear. I just realized that if the user is passing `--dotenv-path` or `--dotenv-filename`, that should probably override `set dotenv-load := false`.

So this should be something like:

```suggestion
  if !settings.dotenv_load.unwrap_or(true) && config.dotenv_filename.is_none() && config.dotenv_path.is_none() {
```

This should also have tests for both cases, if you have a justfile with `set dotenv-load := false`, passing either flag causes an environment file to be loaded.